scheduler/scheduler-server/src/main/java/org/ow2/proactive/threading/TimeoutThreadPoolExecutor.java

```diff
@@ -45,13 +38,31 @@ public TimeoutThreadPoolExecutor(int corePoolSize, int maximumPoolSize, long kee
         super(corePoolSize, maximumPoolSize, keepAliveTime, unit, workQueue, threadFactory);
     }
 
-    public static TimeoutThreadPoolExecutor newFixedThreadPool(int nThreads, ThreadFactory threadFactory) {
-        return new TimeoutThreadPoolExecutor(nThreads,
-                                             nThreads,
-                                             0L,
-                                             TimeUnit.MILLISECONDS,
-                                             new LinkedBlockingQueue<Runnable>(),
-                                             threadFactory);
+    public static TimeoutThreadPoolExecutor newCachedThreadPool(int nThreads, ThreadFactory threadFactory) {
+
+        BlockingQueue<Runnable> queue = new LinkedTransferQueue<Runnable>() {
+            @Override
+            public boolean offer(Runnable e) {
+                return tryTransfer(e);
+            }
+        };
+        TimeoutThreadPoolExecutor threadPool = new TimeoutThreadPoolExecutor(0,
+                                                                             nThreads,
+                                                                             120L,
```

Comment:
Does this parameter keepAliveTime (`120L`) need to be configurable? Or put it as a public static constant if it might be changed in the future?

Reply:
@tao-xinxiu 

I have used the same configuration in various other thread pools, without making the keepAliveTime configurable.

It could be configurable yes, but I would need to change it in various places.